src/gui/editors/PianoRoll.cpp

```diff
@@ -2776,6 +2826,121 @@ void PianoRoll::updateKnifePos(QMouseEvent* me)
 	m_knifeTickPos = mouseTickPos;
 }
 
+void PianoRoll::setupSelectedChords()
+{
+	//
+	// Setup chords
+	//
+	// A chord is an island of notes--as the loop goes over the notes, if the notes overlap,
+	// they are part of the same chord. Else, they are part of a new chord.
+	//
+
+	m_selectedChords.clear();
+	m_midiClip->rearrangeAllNotes();
+
+	const NoteVector& selectedNotes = getSelectedNotes();
+	if (selectedNotes.empty()) { return; }
+
+	int maxTime = -1;
+	NoteVector currentChord;
+	for (Note* note: selectedNotes)
+	{
+		// If the note is not in the current chord range (and this isn't the first chord), start a new chord.
+		if (note->pos() >= maxTime && maxTime != -1)
+		{
+			// Sort the notes by key before adding the chord to the vector
+			std::sort(currentChord.begin(), currentChord.end(), [](Note* a, Note* b){ return a->key() < b->key(); });
+			m_selectedChords.push_back(currentChord);
+			currentChord.clear();
+			maxTime = note->endPos();
+		}
+		maxTime = std::max(maxTime, static_cast<int>(note->endPos()));
+		currentChord.push_back(note);
+	}
+	// Add final chord
+	std::sort(currentChord.begin(), currentChord.end(), [](Note* a, Note* b){ return a->key() < b->key(); });
+	m_selectedChords.push_back(currentChord);
+}
+
+void PianoRoll::updateStrumPos(QMouseEvent* me, bool initial, bool warp)
+{
+	// Calculate the TimePos from the mouse
+	int mouseViewportPos = me->x() - m_whiteKeyWidth;
+	int mouseTickPos = mouseViewportPos * TimePos::ticksPerBar() / m_ppb + m_currentPosition;
+	// Should we add quantization? probably not?
+	if (initial)
+	{
+		m_strumStartTime = mouseTickPos;
+		m_strumStartVertical =  me->y();
+	}
+	m_strumCurrentTime = mouseTickPos;
+	m_strumCurrentVertical =  me->y();
+	int strumTicksHorizontal = m_strumCurrentTime - m_strumStartTime;
+	float strumPower = fastPow10f(0.01f * (m_strumCurrentVertical - m_strumStartVertical));
+
+	if (initial)
+	{
+		m_midiClip->addJournalCheckPoint();
+
+		Note* clickedNote = noteUnderMouse();
+		if (clickedNote == nullptr) { return; }
+
+		for (NoteVector chord: m_selectedChords)
+		{
+			for (Note* note: chord)
+			{
+				// Save the current note position
+				note->setOldPos(note->pos());
+				// if this is the clicked note, calculate it's ratio up the chord
+				if (note == clickedNote && chord.size() > 1)
+				{
+					m_strumHeightRatio = 1.f * std::distance(chord.begin(), std::find(chord.begin(), chord.end(), clickedNote)) / (chord.size() - 1);
+				}
+			}
+		}
+	}
+
+	//
+	// Perform the Strum
+	//
+	// Note above the clicked note (relative to each chord) will be strummed down, notes below will be strummed up.
+	// Holding shift raises the amount of movement to a power, causing the strum to be curved/warped.
+	//
+	for (NoteVector chord: m_selectedChords)
+	{
+		// Don't strum a chord with only one note
+		if (chord.size() <= 1) { continue; }
+		int i = 0;
+		for (Note* note: chord)
+		{
+			float heightRatio = 1.f * i / (chord.size() - 1);
```

Comment:
```suggestion
			float heightRatio = static_cast<float>(i) / static_cast<float>(chord.size() - 1);
```
This doesn't look that simple, you could ignore this

Reply:
Does this improve performance or would the compiler automatically do this?

Reply:
I'm not sure, I think it will result in the same code at the end of the day, but it is more explicit